types/block.go

```diff
@@ -1077,13 +1077,11 @@ func (data *EvidenceData) Hash() tmbytes.HexBytes {
 // ByteSize returns the total byte size of all the evidence
 func (data *EvidenceData) ByteSize() int64 {
 	if data.byteSize == 0 && len(data.Evidence) != 0 {
-		for _, ev := range data.Evidence {
-			pb, err := EvidenceToProto(ev)
-			if err != nil {
-				panic(err)
-			}
-			data.byteSize += int64(pb.Size())
+		pb, err := data.ToProto()
+		if err != nil {
+			panic(err)
 		}
+		data.byteSize = int64(pb.Size())
```

Comment:
Wouldn't it be better to calculate this on-the-fly instead of storing it in `byteSize`? This is going to break if we e.g. append to `EvidenceList` between calls to `ByteSize()`.

Reply:
I cached the byte size to avoid having to convert the evidence data back into protobuf when we want to calculate the size for validation

Reply:
Sure, but we're not able to invalidate the cache whenever the data changes. So this is going to give wrong results if we ever change the data between calls.

Reply:
Potentially but you're not supposed to be changing anything in the block, just validating it